core/src/main/java/com/airbnb/aerosolve/core/function/Function.java

```diff
@@ -26,7 +26,11 @@
 
   void resample(int newBins);
 
-  void smooth(double tolerance);
+  // for function not support smooth, just return 0
+  // for function support smooth, return errAndCoeff
+  double smooth(double tolerance);
+  // for function support smooth, return errAndCoeff percentage
+  double smoothByTolerancePercentage(double tolerancePercentage);
```

Comment:
instead of having two different smoothing functions, it might be better to make the percentage an input parameter in `smooth`, since this is in function interface, all the functions would have to implement this even when it's not used